react/Search.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { FC, useState, useEffect, Fragment } from 'react'
+import { Link, Card } from 'vtex.styleguide'
+import { cseSearch } from './clients/cse'
+
+
+interface Props {
+  query: {
+    q: string
+  }
+}
+
+interface SearchResult {
+  title: string,
+  snippet: string,
+  link: string
+}
+
+const Search: FC<Props> = ({ query }) => {
+  const [results, setResults] = useState<SearchResult[]>([])
+  const queryString = query.q || ''
+  useEffect(() => {
+    (
+      async () => {
+        setResults(await cseSearch(queryString))
+      })()
+  }, [query])
+  return (
```

Comment:
This bit of code here is too bloated, space things out for better readability.
Also, it looks a bit weird regarding linting and formatting, did `yarn lint` not complain about this?

Reply:
Also, I'll comment the same thing in your client file, but I really think it would be a better idea to move the search logic to `docs-graphql`, then just query it in the front-end

Reply:
Can you be more specific about what you think that the code needs improving in formatting? I've tried passing the async part to be a separate function defined inside the `Search` component. But as it is an async function, it isn't possible,